src/metabase/automagic_dashboards/rules.clj

```diff
@@ -61,13 +62,13 @@
 (def ^:private FieldType (s/either (s/constrained s/Str ga-dimension?)
                                    (s/constrained s/Keyword field-type?)))
 
-(def ^:private FieldSpec (s/either [FieldType]
+(def ^:private AppliesTo (s/either [FieldType]
```

Comment:
Not needed before merge, but we should switch away from `s/either` as it's deprecated.

Reply:
I know, and I hate them for it. For a large class of specs it means: go write a predicate function for the entire thing.